### DISCUSSION TITLE
RPI4b database migration

### DISCUSSION THREAD

**Discussion by GWONG3** [photoprism/photoprism#1969] [General]
Hi there,

I am completely new to this so please bare with me for asking stupid questions.

I have recently setup Photoprism on my raspberry pi4 using Docker-compose according to the setup guide; having storage and originals on my home shared NTFS formatted drive; but database (installed as default) locally on the raspberry micro sdcard.  Everything works fine.

I have over 300K of pictures so after almost 5 days it has indexed only 1% of the pics and there are many to go and potentially going to take months; in addition, since it is booting off from a small micro sdcard so the micro sd card will probably be out of space due to database size increases.

My current docker-compose.yml is like this:

 volumes:

- "/home/pi/mySharedDrive/Albums/2021:/photoprism/originals/2021"

- "/home/pi/mySharedDrive/photoprism/storage:/photoprism/storage"

mariadb:

restart: unless-stopped

image: arm64v8/mariadb:10.6 # this mariadb image runs on ARM64-based device$

security_opt:

- seccomp:unconfined

- apparmor:unconfined

command: mysqld --innodb-buffer-pool-size=128M --transaction-isolation=READ$

volumes:

- "./database:/var/lib/mysql" # Important, don't remove

So what I want to do is moving the entire setup from my RPI to my MacBook Air M1 which has much more processing power and memory/storage, complete the indexing of these 300K of pics; and move everything back to RPI (will get a bigger SDCARD once I reach that point to store the database).  Can I do that?  I guess I need step by step instructions how to migrate the database from the raspberry PI4 to my MacBook Air M1 which has Docker-compose running but running MariaDB in a container; and the steps how to migrate the database from MacBook Air M1 back to RPI once the indexing is finished.

Thanks in advance.

Regards,

GW

**Comment by tonyhouse1982** [comment]
You can indexing all inside macbook
then export database (mysqldump) and photoprism folders
and import back into pi

**Reply by GWONG3** [reply]
Thanks, yes I have done exactly the same as you described a few days ago and it works!
